posts/inside-rust/2024-06-19-this-development-cycle-in-cargo-1.80.md

```diff
@@ -0,0 +1,517 @@
+Separate from that Call for Testing, we did receive feedback about the
+`latest` annotations in `cargo update` ([#13908](https://github.com/rust-lang/cargo/issues/13908)).
+Discussion is still on-going.
+
+<!-- 2024-06-04 -->
+As far as we are aware, the last remaining step before stabilization is
+renaming the configuration ([#13540](https://github.com/rust-lang/cargo/issues/13540)):
+```toml=
+[resolver]
+something-like-precedence = "something-like-rust-version"
+```
+
+We want to keep in mind how this might align with future possibilities like:
+- [#5657 Minimal version resolution](https://github.com/rust-lang/cargo/issues/5657)
+- [#4225 Control of yanked](https://github.com/rust-lang/cargo/issues/4225)
+- [#13290 Control of prerelease](https://github.com/rust-lang/cargo/issues/13290)
+
+For minimal vs maximal version resolution, we see that is likely an enumerate value under an unnamed key, so we can set that aside.
+
+For the rest, we are talking about the following modes:
+| mode                                 | MSRV     | yanked   | prerelease |
+|--------------------------------------|----------|----------|----------|
+| this is yet another candidate        | Required | Never?   | Never?   |
+| de-priorize this over other versions | Required | Likely   | Likely   |
+| don't resolve to if already in use   | Likely   | Required | Required |
+
+This helped to show that we probably want to name the field `incompatible-rust-version` to clarify that we are talking about how we are handling those packages and to leave room for `resolver.rust-version` to override what version is used when resolving dependencies.
+
+The challenge is providing a clear way to communicate the "mode".
+For de-prioritizing, we considered:
+- discourage
+- eschew
+- avoid
+- fallback
+
+Of those, `fallback` most captures the nuance but that is a noun while we had been considering `allow` and `deny` for the other modes which are verbs.
+
+We had considered a short-term solution of a "precedence policy" field that would take a policy name, much like we have profile names that encompass many compiler settings.
+This would be short term and could run into its own issues with being confusing, so we decided to focus on the "right" solution.
+
+##### Removing implicit features
+
+*Update from [1.78](https://blog.rust-lang.org/inside-rust/2024/03/26/this-development-cycle-in-cargo-1.78.html#weak-feature-syntax) and [1.79](https://blog.rust-lang.org/inside-rust/2024/05/07/this-development-cycle-in-cargo-1.79.html#user-controlled-cargo-diagnostics)*
+
+A crater run exposed a couple of bugs with `cargo fix` when migrating implicit features to explicit features ([#14010](https://github.com/rust-lang/cargo/issues/14010)).
+
+<!-- 2024-06-11 -->
+This exposed a misunderstand that `["dep_name/feature_name"]` is not always the same as `["dep:dep_name", "dep_name?/feature_name"]` because the former won't suppress the creation of an implicit feature.
```

Comment:
Perhaps a typo:
```suggestion
This exposed a misunderstanding that `["dep_name/feature_name"]` is not always the same as `["dep:dep_name", "dep_name?/feature_name"]` because the former won't suppress the creation of an implicit feature.
```